x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistryMetadata.java

```diff
@@ -260,7 +274,9 @@ public boolean equals(Object obj) {
             return false;
         }
         ModelRegistryMetadata other = (ModelRegistryMetadata) obj;
-        return Objects.equals(this.modelMap, other.modelMap) && isUpgraded == other.isUpgraded;
+        return Objects.equals(this.modelMap, other.modelMap)
+            && isUpgraded == other.isUpgraded
+            && Objects.equals(this.tombstones, other.tombstones);
```

Comment:
This indicates that the `mutateInstance()` method in `ModelRegistryMetadataTests` is not properly implemented, because it should have caught that `tombstones` were not being included in the `equals()` method. Would it be possible to fix it?